feature/gnmi/gnmi_subscriptionlist_test/README.md

```diff
@@ -0,0 +1,51 @@
+# GNMI-2: gnmi_subscriptionlist_test
+
+## Summary
+This is to test for gNMI `Subscription` to multiple paths with different `SubscriptionMode` in a single `SubscriptionRequest` message using the `Subscriptionlist` field. Goal here is to,
+  * Ensure that the NOS supports "Subscriptionlist" field with multiple `Subscription` messages and also supports the desired `Subscriptionmode` per path in each `Subscription` message.
+  * The tests also check if the DUT is responding back everytime with a `SubscriptionResponse` message that has the `sync_response` field set to `true`
+
+## Procedure
+### GNMI-2.1: Verify single subscription request with a Subscriptionlist and different SubscriptionModes:
+  * Send a single `SubscribeRequest` message to the DUT with a **SubcriptionList** and **SubscriptionMode** matching the "Telemetry Parameter Coverage" section below. Use `Stream` mode for the `SubcribeRequest`.
+  * Ensure that the implementation successfully allows subscription to all the paths mentioned below and a `SubscribeResponse` message is received by the client with the `sync_reponse` field set to `true`. The RPC via which the `SubscribeRequest` was recieved should eventually be closed by the client.
+### GNMI-2.2: Change SubscriptionModes in the subscription list and verify receipt of sync_response:
+  * In the "Telemetry Parameter coverage" section below, change the `Subscribe` message for each of the paths with `SubscriptionMode` as `ON_CHANGE` to `TARGET_DEFINED` and the ones that are `TARGET_DEFINED` to `SAMPLE` and send all the subscribe messages in a single `SubscribeRequest` message to the DUT. Confirm that a `SubscribeResponse` message is received by the client with the `sync_reponse` field set to `true`. The client should then close the RPC session
```

Comment:
What sample interval should be used?  Other tests are specifying two.  10 second and 15 second.  Probably use that for now.  (can tweak it later if needed)